Doc alternative space array package (#160)
JuliaSparse/SparseArrays.jl: docs/src/index.md

```

Julia has support for sparse vectors and [sparse matrices](https://en.wikipedia.org/wiki/Sparse_matrix)

> in the `SparseArrays` stdlib module, but limted to (1D and) 2D arrays, unlike the regular dense arrays.
> There's an external package [SparseArrayKit.jl](https://github.com/Jutho/SparseArrayKit.jl) available
> for multidimensional sparse arrays (and also other packages for sparse).
> 
> Sparse arrays are arrays that contain enough zeros that storing them in a special data structure leads
> to savings in space and execution time, compared to dense arrays.

Reviewer: The first line (sparse vectors and sparse matrices) implies 1D and 2D. 
```suggestion
in the `SparseArrays` stdlib module. Sparse arrays are arrays that contain enough zeros that storing them in a special data structure leads to savings in space and execution time, compared to dense arrays.

External packages which implement different sparse storage types, multidimensional sparse arrays, and more can be found in [Noteworthy external packages](@ref man-csc)
```

3. [SparseMatricesCSR.jl](https://github.com/gridap/SparseMatricesCSR.jl) provides a Julia native implementation of the Compressed Sparse Rows (CSR) format.

4. [MKLSparse.jl](https://github.com/JuliaSparse/MKLSparse.jl) accelerates SparseArrays sparse-dense matrix operations using Intel's MKL library.

5. [SparseArrayKit.jl](https://github.com/Jutho/SparseArrayKit.jl) available for multidimensional sparse arrays.

> 6. [LuxurySparse.jl](https://github.com/QuantumBFS/LuxurySparse.jl)

Reviewer: ```suggestion
6. [LuxurySparse.jl](https://github.com/QuantumBFS/LuxurySparse.jl) provides static sparse array formats, as well as a coordinate format.
```

> 7. [ExtendableSparse.jl](https://github.com/j-fu/ExtendableSparse.jl)

Reviewer: ```suggestion
7. [ExtendableSparse.jl](https://github.com/j-fu/ExtendableSparse.jl) enables fast insertion into sparse matrices using a lazy approach to new stored indices.
```
